File: 2679/solution.go
package _2679

import "sort"

func matrixSum(nums [][]int) int {
	res := 0
	m := len(nums)
	n := len(nums[0])
	for i := 0; i < m; i++ {
		sort.Ints(nums[i])
	}
	for j := 0; j < n; j++ {
		maxVal := 0
		for i := 0; i < m; i++ {
			if nums[i][j] > maxVal {
				maxVal = nums[i][j]
			}
		}
		res += maxVal
	}
	return res
}

File: 2679/solution.py
class Solution:
    def matrixSum(self, nums: List[List[int]]) -> int:
        if not nums:
            return 0
        for i in range(len(nums)):
            nums[i].sort()
        res = 0
        for i in range(len(nums[0])):
            tmp = 0
            for j in range(len(nums)):
                tmp = max(tmp, nums[j][i])
            res += tmp
        return res
